src/upf_network.py

```diff
@@ -115,7 +171,50 @@ def unset_ip_address(self) -> None:
             logger.info("Removing IP %s from interface %s", self.ip_address, self.name)
             iface_record.del_ip(self.ip_address).commit()
             return
-        logger.warning("Interface %s not found in the network database", self.name)
+        logger.warning(
+            "Unsetting IP for interface %s failed: Interface not found in the network database",
+            self.name,
+        )
+
+    def create(self) -> None:
+        """Create given network interface.
+
+        In DPDK mode the actual network interfaces of the host are removed from the management
+        of the kernel. To be able to handle ICMP and ARP requests UPF uses virtual interfaces.
+        This method creates a virtual interface with the MAC address matching the corresponding
+        physical interface and tags it with the PCI address of the physical interface.
+        """
+        peer_interface_name = f"{self.name}-vdev"
+        (self.network_db.interfaces.create(  # type: ignore[reportAttributeAccessIssue]
+            ifname=self.name,
+            kind="veth",
+            peer={"ifname": peer_interface_name},
+        ).set(
+            state="up",
+            address=self.mac_address,
+            mtu=self.mtu_size,
+        ).add_ip(address=self.ip_address, prefixlen=24).commit())
+        self.network_db.reload()
+        self.network_db.interfaces[self.name].set(ifalias=self.alias).commit()  # type: ignore[reportAttributeAccessIssue]  # noqa: E501
+        self.network_db.interfaces[peer_interface_name].set(state="up").commit()  # type: ignore[reportAttributeAccessIssue]  # noqa: E501
```

Comment:
There are many static analysis ignores here, do we need those? We should probably add some validations instead

Reply:
The reason why we're getting those is that the `NDB` class doesn't have the `interfaces` field. I think this is related to the design of NDB which creates objects for network entities upon request. 
So far I didn't found any solution to that problem.